hoard/models.py

```diff
@@ -1,16 +1,139 @@
+from typing import Any, Dict, List, Optional
+
 import attr
 
 
+@attr.s(auto_attribs=True)
```

Comment:
One thing that might be worth considering is adding an `asdict()` method to each of these submodels. This method would output the Dataverse representation for each one. This would remove the need to have separate functions to build each part of the Dataset model and also simplify the `asdict()` method of the Dataset.

Reply:
Might come back to this for a later PR.